### PR TITLE
Miscellanous small improvements

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

* Avoid repeated regex compilation that was slowing down mechanism import
* Switch high-debug-level flame solver output from XML to YAML
* Remove unused variable OneDim::m_solve_time
* Use ruamel.yaml instead of the out-of-date ruamel_yaml
* Add Mu0Poly and Nasa9PolyMultiTempRegion to Python docs
* Improve error messages from `fpValueCheck`
* Fix a bunch of miscellaneous spelling errors

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: src/base/stringUtils.cpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ doublereal fpValueCheck(const std::string& val)
     ch = str[0];
     if (ch == '+' || ch == '-') {
         if (str.size() == 1) {
-            throw CanteraError("fpValueCheck", "string ends in '{}'", ch);
+            throw CanteraError("fpValueCheck", "string '{}' ends in '{}'", val, ch);
```
Comment (ischoegl) [inline src/base/stringUtils.cpp:150]: Would it make sense to start using f-strings in those instances?

**Review by ischoegl — COMMENTED** [review]
Thank you, @speth - this covered a lot of ground! I only have a minor comment.

File: src/base/stringUtils.cpp (hunk 1 above)
Comment (speth) [reply src/base/stringUtils.cpp:150]: What? This is in C++...

Comment (ischoegl) [reply src/base/stringUtils.cpp:150]: oops! nevermind ... apparently coded too much Python lately. 

In that case, this all looks fine to me!!

**Review by ischoegl — APPROVED** [review]

**Pull request merged**